back-end/controllers/record-controller.js

```diff
@@ -22,104 +22,118 @@ const recordController = {
-      const userIds = await prisma.$queryRaw`SELECT id FROM "User";`
-      const userId = userIds.map(({ id }) => id)[0]
-
-      // Get categoryIds
-      const result =
-        await prisma.$queryRaw`SELECT id FROM "Category" WHERE NOT "mainCategory" = 'Income' AND NOT "mainCategory" = 'Savings';`
-      const categoryIds = result.map(({ id }) => id)
-
-      // Basic query object, without query string
-      // const prismaQuery = {
-      //   where: {
-      //     userId: 33,
-      //   },
-      // }
-
-      // With query string
-
-      // Get the current user's all expense records
-      // const expenseRecords = await prisma.record.findMany(prismaQuery)
-      const expenseRecords =
-        await prisma.$queryRaw`SELECT * FROM "Record" WHERE "userId" = ${userId} AND "categoryId" IN (${Prisma.join(
-          categoryIds
-        )});`
-
-      res.json(expenseRecords)
     } catch (error) {
       next(error)
     }
   },
 
-  // edit a record
+  // Edit the record
+  // URL: put /records/:rid
   putRecord: async (req, res, next) => {
     try {
-      // #swagger.tags = ['Expense Record']
       const recordId = Number(req.params.rid)
       const { title, amount, note, categoryId } = req.body
-      // uncommend after adding authentication process
-      // const userId = req.user.id
+      const userId = req.user.id
 
+      // Check if there are missing data
       if (!title || !amount || !categoryId)
         return res.json(
           'missing title or amount or category id to update this record'
         )
 
+      // Check if the record is in database
       const theRecord = await prisma.record.findUnique({
         where: {
           id: recordId,
         },
       })
       theRecord ? theRecord : res.json('the record does not exist')
 
+      // Check if the record is one of the current user's records
+      userId === theRecord.userId
+        ? userId
+        : res.json('this record does not belong to the current user')
+
+      // Update this record
       await prisma.record.update({
         where: { id: recordId },
-        data: { title, amount, note, categoryId, userId: 13 },
+        data: { title, amount, note, categoryId },
       })
+
       res.json('successfully update this record')
+      // #swagger.tags = ['Expense Record']
     } catch (error) {
       next(error)
     }
   },
 
-  // delete a expense record
+  // Delete the record
+  // URL: delete /records/:rid
   deleteRecord: async (req, res, next) => {
     try {
-      // #swagger.tags = ['Expense Record']
       const recordId = Number(req.params.rid)
 
+      // Check if the record is in database
       const theRecord = await prisma.record.findUnique({
         where: {
           id: recordId,
         },
       })
       theRecord ? theRecord : res.json('the record does not exist')
 
+      // Check if the record is one of the current user's records
+      userId === theRecord.userId
```

Comment:
It seems you forget to declare userId in this controller before assiging to it.

Reply:
I have added taking `userId` from `req` on the latest commit.